src/app/utils/utils.js

```diff
@@ -178,27 +178,54 @@ const getIdentifierQuery = (identifierNumbers = {}) =>
     .map(([key, value]) => (value ? `&${key}=${value}` : ''))
     .join('');
 
+/**
+ * adobeAnalyticsQueryString
+ * Utility function that builds a query string as expected by the Adobe Analytics dashboard
+ * @param {string} param value of param to be passed into the query string field.
+ */
+const adobeAnalyticsQueryString = (param = "") => {
+  return param.length ? `[|${param}]` : "";
+}
+
+/**
+ * adobeAnalyticsParam
+ * Utility function that builds a param string as expected by the Adobe Analytics dashboard
+ * @param {string} param value of param to be passed into the param string field.
+ */
+const adobeAnalyticsParam = (param = "") => {
+  return param.length ? `|[${param}]` : "";
+}
+
 // Maps routes to the appropriate page name for Adobe Analytics.
-const adobeAnalyticsRouteToPageName = (route = '') => {
+const adobeAnalyticsRouteToPageName = (route = '', queryParams = '')=> {
+  // parse additional route attributes
+  let bnumber = route.includes('/bib') ? route.split('/')[2] : "";
+  bnumber = bnumber && standardizeBibId(bnumber);
+
+  const holdItem = route.includes('/hold') ? route.split("/")[3] : "";
+  const holdBibAndItem = holdItem && holdItem.length && holdItem.split("-");
+
+  const uuid = route.includes('/subject_headings') ? route.split("/")[2] : "";
+
   switch (route) {
     case route.match(/\/search\/advanced/i)?.input:
       return ADOBE_ANALYTICS_PAGE_NAMES.ADVANCED_SEARCH;
     case route.match(/\/search/i)?.input:
-      return ADOBE_ANALYTICS_PAGE_NAMES.SEARCH_RESULTS;
+      return `${ADOBE_ANALYTICS_PAGE_NAMES.SEARCH_RESULTS}${adobeAnalyticsQueryString(queryParams)}`;
     case route.match(/\/bib(\/[^\/]*)\/all/i)?.input:
-      return ADOBE_ANALYTICS_PAGE_NAMES.DETAILS_ALL_ITEMS;
+      return `${ADOBE_ANALYTICS_PAGE_NAMES.BIB}${adobeAnalyticsParam(bnumber)}|all`;
     case route.match(/\/bib/i)?.input:
-      return ADOBE_ANALYTICS_PAGE_NAMES.DETAILS;
+      return `${ADOBE_ANALYTICS_PAGE_NAMES.BIB}${adobeAnalyticsParam(bnumber)}`;
     case route.match(/\/hold\/request(\/[^\/]*)\/edd/i)?.input:
-      return ADOBE_ANALYTICS_PAGE_NAMES.EDD_REQUEST; env
+      return `${ADOBE_ANALYTICS_PAGE_NAMES.EDD_REQUEST}${adobeAnalyticsParam(holdBibAndItem[0])}${adobeAnalyticsParam(holdBibAndItem[1])}`;
```

Comment:
Maybe be more explicit and assign [0] and [1] to variables for bnumber and inumber.